appdata: Improve appdata for AppStream 1.0

sabriunal opened the pull request:
### appdata: Improve appdata for AppStream 1.0]

- Add the `<developer><name>` tag
- Mark the `<developer_name>` tag as deprecated
- Update display_length tag
- Fix screenshot tags
- Update appstreamcli validation

### appdata: Fix my mistake

"Icons and categories

If there’s a type="desktop-id" launchable, they get pulled from it.
Most of the icon not found errors with the flathub builder can be
traced down to the launchable value not matching the desktop file name.

Don’t set them in the AppData unless you want to override them
(even though then it might be a better idea to patch the desktop file
itself)."

More information: https://docs.flathub.org/docs/for-app-authors/appdata-guidelines/#icons-and-categories

ADBeveridge commented:
I got the pr request order wrong, can you fix the conflict please? Thanks @sabriunal .

yakushabb committed:
appdata: Remove categories

"Icons and categories

If there’s a type="desktop-id" launchable, they get pulled from it.
Most of the icon not found errors with the flathub builder can be
traced down to the launchable value not matching the desktop file name.

Don’t set them in the AppData unless you want to override them
(even though then it might be a better idea to patch the desktop file
itself)."

More information: https://docs.flathub.org/docs/for-app-authors/appdata-guidelines/#icons-and-categories

yakushabb committed:
appdata: Improve appdata for AppStream 1.0

- Add the `<developer><name>` tag
- Mark the `<developer_name>` tag as deprecated
- Update display_length tag
- Fix screenshot tags
- Update appstreamcli validation

ghost commented:
Updated.
